tests/integration/server/proposal_api_tests.py

```diff
@@ -0,0 +1,378 @@
+            manager_payload["manager"] = manager_id
+            manager_id = create_user(session, manager_payload)
+            old_expected_approver_ids.append(manager_id)
+            old_usernames_for_deleting.append(manager_payload["username"])
+            i += 1
+        roleowner_payload["manager"] = manager_id
+        roleowner_id = create_user(session, roleowner_payload)
+        old_expected_approver_ids.append(roleowner_id)
+        role_payload = {"name": "Test_role"}
+        role_payload["owners"] = (roleowner_id,)
+        role_payload["administrators"] = (roleowner_id,)
+        role_id = create_role(session, role_payload)
+        user_payload = {
+            "name": "user_name",
+            "username": "user_username",
+            "password": "user_password",
+            "email": "user@email_id",
+        }
+        user_id = create_user(session, user_payload)
+        creating_proposal_payload = {"id": user_id, "reason": "add the test member"}
+        proposal_id = create_proposal(session, creating_proposal_payload, role_id)
+        proposal_information = get_proposal_details(session, proposal_id)
+        i = 0
+        while i < len(old_expected_approver_ids):
+            assert (
+                old_expected_approver_ids[i]
+                in proposal_information.json()["data"]["approvers"]
+            )
+            i += 1
+        manager_payload = create_test_manager_payload()
+        new_expected_approver_ids = []
+        new_usernames_for_deleting = []
+        new_usernames_for_deleting.append(manager_payload["username"])
+        manager_id = create_user(session, manager_payload)
+        new_expected_approver_ids.append(manager_id)
+        i = 0
+        while i < 4:
+            manager_payload = create_test_manager_payload()
+            manager_payload["manager"] = manager_id
+            manager_id = create_user(session, manager_payload)
+            new_expected_approver_ids.append(manager_id)
+            new_usernames_for_deleting.append(manager_payload["username"])
+            updated_manager_id = manager_id
+            i += 1
+        conn = connect_to_db()
+        r.table("users").filter({"next_id": roleowner_id}).update(
+            {"manager_id": updated_manager_id}
+        ).run(conn)
+        conn.close()
+        new_expected_approver_ids.append(roleowner_id)
+        user_payload = {
+            "name": "user_name1",
+            "username": "user_username1",
+            "password": "user_password1",
+            "email": "user1@email_id",
+        }
+        user_id = create_user(session, user_payload)
+        creating_proposal_payload = {"id": user_id, "reason": "add the test member"}
+        proposal_id = create_proposal(session, creating_proposal_payload, role_id)
+        proposal_information = get_proposal_details(session, proposal_id)
+        i = 0
+        while i < len(new_expected_approver_ids):
+            assert (
```

Comment:
As we discussed please try to write the tests in a way that doesn't require an assertion in a while loop and preferably has one assertion per function.